src/test/fakeSuite/extension.test.ts

```diff
@@ -1,286 +1,330 @@
+            // from this extension repository instead of a real VS installation that happens to be in the path.
+            configuration.setPreConfigureScript(path.join(vscode.workspace.rootPath || "./", ".vscode/preconfigure.bat"));
+            await make.preConfigure(make.TriggeredBy.tests);
+
+            configuration.prepareConfigurationsQuickPick();
+            configuration.setConfigurationByName("InterestingSmallMakefile_windows_configDebug");
+            const retc: number = await make.cleanConfigure(make.TriggeredBy.tests, true);
+
+            configuration.setBuildBeforeLaunch(false);
+            const launchConfigurations: string[] = ["bin\\InterestingSmallMakefile\\ARC H3\\Debug\\main.exe(str3a,str3b,str3c)",
+                                                    "bin\\InterestingSmallMakefile\\arch1\\Debug\\main.exe(str3a,str3b,str3c)",
+                                                    "bin\\InterestingSmallMakefile\\arch2\\Debug\\main.exe()"];
+            for (const config of launchConfigurations) {
+                await configuration.setLaunchConfigurationByName(vscode.workspace.rootPath + ">" + config);
+                let status: string = await launch.getLauncher().validateLaunchConfiguration(make.Operations.debug);
+                let launchConfiguration: configuration.LaunchConfiguration | undefined;
+                if (status === launch.LaunchStatuses.success) {
+                    launchConfiguration = configuration.getCurrentLaunchConfiguration();
+                }
+
+                if (launchConfiguration) {
+                    launch.getLauncher().prepareDebugCurrentTarget(launchConfiguration);
+                    launch.getLauncher().prepareRunCurrentTarget();
+                }
+            }
+
+            // A bit more coverage, "RelSize" and "RelSpeed" are set up
+            // to exercise different combinations of pre-created build log and/or make tools.
+            // No configure is necessary to be run here, it is enough to look at what happens
+            // when changing a configuration.
+            configuration.setConfigurationByName("InterestingSmallMakefile_windows_configRelSize");
+            configuration.setConfigurationByName("InterestingSmallMakefile_windows_configRelSpeed");
+
+            // InterestingSmallMakefile_windows_configRelSpeed constructs a more interesting build command.
+            configuration.setTargetByName("Execute_Arch3");
+            make.prepareBuildTarget("Execute_Arch3");
+
+            // Compare the output log with the baseline
+            // TODO: incorporate relevant diff snippets into the test log.
+            // Until then, print into base and diff files for easier viewing
+            // when the test fails.
+            let parsedPath: path.ParsedPath = path.parse(extensionLogPath);
+            let baselineLogPath: string = path.join(parsedPath.dir, "../InterestingSmallMakefile_windows_baseline.out");

```

Comment:
I don't understand the `..` here.  I thought the logs and .outs were in the same folder.

Reply:
It's just how I set it up. I defined the extension log to be put in the .vscode of the fake repro since it's something that is always regenerated. It was only habit. If I changed that location then I wouldn't need the "..", indeed.

Reply:
ok